### PR TITLE
Add breaking improvements page

### PR DESCRIPTION
**Pull request by jribbink**
https://github.com/onflow/docs/issues/674

### CODE REVIEW

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 1]
```diff
@@ -0,0 +1,1650 @@
...
+
+## 💫 New features
+
+<details>
+
+<summary>View Functions added</summary>
+
+#### 💡 Motivation
+
+View functions allow developers to improve the reliability and safety of their programs, and helps them to reason about the effects of their and the programs of others.
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:15]:
```suggestion
View functions enable developers to enhance the reliability and safety of their programs, facilitating a clearer understanding of the impacts of their own code and that of others.
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 2]
```diff
@@ -0,0 +1,1650 @@
...
+
+Many of the improvements of Cadence 1.0 are fundamentally changing how Cadence works and how it is used. However, that also means it is necessary to break existing code to release this version, which will guarantee stability (no more planned breaking changes) going forward.
+
+Once Cadence 1.0 is live, breaking changes will simply not be acceptable.
+
+So we have, and need to use, this last chance to fix and improve Cadence, so it can deliver on its promise of being a language that provides security and safety, while also providing composability and simplicity.
+
+We very much understand that it is painful for developers to have their code get broken and require them to update it.
+
+However, we believe that the pain is worth it, given the significant improvements that make Cadence development significantly more powerful and pleasant, and enabling developers to for write and deploy immutable contracts.
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:134]:
```suggestion
We fully recognize the frustration developers feel when updates break their code, necessitating revisions. Nonetheless, we are convinced that this inconvenience is justified by the substantial enhancements to Cadence development. These improvements not only make development more effective and enjoyable but also empower developers to write and deploy immutable contracts.
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 3]
```diff
@@ -0,0 +1,1650 @@
...
+
+#### ℹ️ Description
+
+Pre-conditions and post-conditions are now considered `view` contexts, meaning that any operations that would be prevented inside of a `view` function are also not permitted in a pre-condition or post-condition.
+
+This is to prevent underhanded code wherein a user modifies global or contract state inside of a condition, where they are meant to simply be asserting properties of that state.
+
+In particular, since only expressions were permitted inside conditions already, this means that if users wish to call any functions in conditions, these functions must now be made `view` functions.
+
+This improvement was proposed in [FLIP 1056 7](https://github.com/onflow/flips/blob/main/cadence/20220715-cadence-purity-analysis.md). To learn more, please consult the FLIP and documentation.
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:156]:
```suggestion
This improvement was proposed in [FLIP 1056](https://github.com/onflow/flips/blob/main/cadence/20220715-cadence-purity-analysis.md). To learn more, please consult the FLIP and documentation.
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 4]
```diff
@@ -0,0 +1,1650 @@
...
+```cadence
+let number = 1
+let ref: &Int = &number
+```
+
+</details>
+
+<details>
+
+<summary>Naming Rules Got Tightened</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:360]:
```suggestion
<summary>Tightening Of Naming Rules</summary>
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 5]
```diff
@@ -0,0 +1,1650 @@
...
+
+Here, the `access(Withdraw)` syntax means that a reference to `Vault` must possess the `Withdraw` entitlement in order to be allowed to call the `withdraw` function, which can be given when a reference or `Capability` is created by using a new syntax: `auth(Withdraw) &Vault`.
+
+This would allow developers to safely downcast `&{Provider}` references to `&Vault` references if they want to access functions like `deposit` and `balance`, without enabling them to call `withdraw`.
+
+</details>
+
+<details>
+
+<summary>`pub` and `priv` Access Modifiers Got Removed</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:652]:
```suggestion
<summary>Removal of `pub` and `priv` Access Modifiers</summary>
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 6]
```diff
@@ -0,0 +1,1650 @@
...
+Additionally, because what fields and functions were available to a reference depended on what the type of the reference was, references could not be downcast, leading to ergonomic issues.
+
+#### ℹ️ Description
+
+Access control has improved significantly.
+When giving another user a reference or `Capability` to a value you own, the fields and functions that the user can access is determined by the type of the reference or `Capability`.
+
+Previously, access to a value of type `T`, e.g. via a reference `&T`, would give access to all fields and functions of `T`. Access could be restricted, by using a restricted type. For example, a restricted reference `&T{I}` could only access members that were `pub` on `I`. Since references could not be downcast, any members defined on `T` but not on `I` were unavailable to this reference, even if they were `pub`.
+
+Access control is now handled using a new feature called Entitlements, as originally proposed across [\*\*FLIP 54 3](https://github.com/onflow/flips/blob/main/cadence/20221214-auth-remodel.md)** and **[FLIP 94 3](https://github.com/onflow/flips/blob/main/cadence/20230623-entitlement-improvements.md)\*\*.
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:550]:
```suggestion
Access control is now handled using a new feature called Entitlements, as originally proposed across **[FLIP 543](https://github.com/onflow/flips/blob/main/cadence/20221214-auth-remodel.md)** and **[FLIP 943](https://github.com/onflow/flips/blob/main/cadence/20230623-entitlement-improvements.md)**.
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 7]
```diff
@@ -0,0 +1,1650 @@
...
+
+  /* ... rest of interface ... */
+}
+```
+
+</details>
+
+<details>
+
+<summary>Restricted Types Got Replaced with Intersection Types</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:721]:
```suggestion
<summary>Replacement of Restricted Types with Intersection Types</summary>
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 8]
```diff
@@ -0,0 +1,1650 @@
...
+
+With the previously mentioned entitlements feature, which uses `access(E)` syntax to denote entitled access, the `pub`, `priv` and `pub(set)` modifiers became the only access modifiers that did not use the `access` syntax.
+
+This made the syntax inconsistent, making it harder to read and understand programs.
+
+In addition, `pub` and `priv` already had alternatives/equivalents: `access(all)` and `access(self)`.
+
+#### ℹ️ Description
+
+The `pub`, `priv` and `pub(set)` access modifiers got removed from the language, in favor of their more explicit `access(all)` and `access(self)` equivalents (for `pub` and `priv`, respectively).
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:664]:
```suggestion
The `pub`, `priv` and `pub(set)` access modifiers are being removed from the language, in favor of their more explicit `access(all)` and `access(self)` equivalents (for `pub` and `priv`, respectively).
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 9]
```diff
@@ -0,0 +1,1650 @@
...
+    .borrow<&{HasCount}>(/public/hasCount)!
+  return counterRef.count
+}
+```
+
+</details>
+
+<details>
+
+<summary>External Mutation Got Improved</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:1303]:
```suggestion
<summary>External Mutation Improvement</summary>
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 10]
```diff
@@ -0,0 +1,1650 @@
...
+// Calling a mutating function
+masterCollectionRef.kittyCollection.deposit(<-nft)
+
+```
+
+</details>
+
+<details>
+
+<summary>Nested Type Requirements Got Removed</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:1440]:
```suggestion
<summary>Removal Of Nested Type Requirements</summary>
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 11]
```diff
@@ -0,0 +1,1650 @@
...
+    view fun getCount(): Int {
+//  ^^^^ addedreturnself.ownedNFTs.length
+    }
+```
+
+</details>
+
+<details>
+
+<summary>Interface Inheritance added</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:76]:
```suggestion
<summary>Interface Inheritance Added</summary>
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 12]
```diff
@@ -0,0 +1,1650 @@
...
+
+#### 💡 Motivation
+
+Previously, interfaces could not inherit from other interfaces, which required developers to repeat code.
+Interface inheritance allows code abstraction and code reuse.
+
+#### ℹ️ Description and ✨ example
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:83]:
```suggestion
#### ℹ️ Description and ✨ Example
```

File: versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md [hunk 13]
```diff
@@ -0,0 +1,1650 @@
...
+}
+```
+
+We would like to thank community member @justjoolz for reporting this bug.
+
+</details>
+
+<details>
+
+<summary>Incorrect Operators in Reference Expressions Get Reported</summary>
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/breaking-improvements.md:295]:
```suggestion
<summary>Incorrect Operators In Reference Expressions Get Reported</summary>
```

**Review by gregsantos — CHANGES_REQUESTED** [review]

**Comment by Aliserag** [comment]
Some nits @jribbink https://www.loom.com/share/caba14407c62436193b9677a34f73e73?sid=a0c519c7-f18d-4d37-ad54-b62f1159b790

**Review by gregsantos — APPROVED** [review]
Great!

**Pull request merged**